tunneld-server/src/server.rs

```diff
@@ -197,22 +194,34 @@ impl TunnelService for Handler {
                 while let Some(connection) = new_connection_rx.recv().await {
                     let connection_id = connection.id.to_string();
                     connections.insert(connection.id.clone(), connection);
-                    tx.send(Ok(Control {
-                        command: Command::Work as i32,
-                        payload: Some(Payload::Work(WorkPayload { connection_id })),
-                    }))
-                    .await
-                    .context("failed to send work command")
-                    .unwrap();
+                    outbound_streaming_tx
```

Comment:
the primary change is here, after this error response, the client will close the streaming, then server doesn't be blocked anymore